src/NServiceBus.Core/Transports/Learning/LearningTransportDispatcher.cs

```diff
@@ -57,6 +59,14 @@ Task DispatchUnicast(IEnumerable<UnicastTransportOperation> operations, Transpor
 
         async Task WriteMessage(string destination, IOutgoingTransportOperation transportOperation, TransportTransaction transaction)
         {
+            var headerPayload = HeaderSerializer.Serialize(transportOperation.Message.Headers);
+            var headerSize = Encoding.UTF8.GetByteCount(headerPayload);
```

Comment:
I was more thinking we should throw if someone gives us a string we cant encode?